integrations/firehydrant/client.py

```
@@ -0,0 +1,156 @@
+
+    @property
+    def api_auth_header(self) -> dict[str, Any]:
+        return {
+            "Authorization": f"{self.api_key}",
+            "Content-Type": "application/json",
+        }
+
+    async def send_api_request(
+        self,
+        endpoint: str,
+        method: str = "GET",
+        query_params: Optional[dict[str, Any]] = None,
+        json_data: Optional[dict[str, Any]] = None,
+    ) -> dict[str, Any]:
+        try:
+            response = await self.http_client.request(
+                method=method,
+                url=f"{self.base_url}/v1/{endpoint}",
+                params=query_params,
+                json=json_data,
+                headers=self.api_auth_header,
+            )
+            response.raise_for_status()
+            return response.json()
+        except httpx.HTTPStatusError as e:
+            logger.error(
+                f"HTTP error with status code: {e.response.status_code} and response text: {e.response.text}"
+            )
+            raise
+
+    async def get_paginated_resource(
+        self, resource_type: str
+    ) -> AsyncGenerator[list[dict[str, Any]], None]:
+        logger.info(f"Getting {resource_type} from Firehydrant")
+        endpoint = endpoint_resource_type_mapper.get(resource_type, resource_type)
+
+        pagination_params: dict[str, Any] = {
+            "page": 1,
+            "per_page": PAGE_SIZE,
+        }
+
+        try:
+            while True:
+                response = await self.send_api_request(
+                    endpoint=endpoint, query_params=pagination_params
+                )
+                if response.get("data"):
+                    yield response["data"]
+                else:
+                    logger.warning(f"No {resource_type} found in the response")
+
+                pagination_params["page"] += 1
+
+                if pagination_params["page"] > response.get("pagination", {}).get(
+                    "pages"
+                ):
```

is it possible for the response to no include pagination & pages?
if the api should retrive those then pls just use a regular `[]` to fetch them and let the error raise from there